src/service/resources-impl.js

```diff
@@ -2411,6 +2432,7 @@ export class Resources {
         } else {
           dev().fine(TAG_, 'document is not visible: no scheduling');
         }
+        this.firstPassDone_.resolve();
```

Comment:
@cramforce is it intended to resolve `firstPassDone` right after `schedulePass` without waiting for the execution?

Reply:
Is that what is happening? Semantically I need it to happen after discoverWork has run once.

Reply:
I was wrong in my previous question.

But still, I guess semantically, this is `firstWorkDone_` instead of `firstPassDone_`. because after this change, any `pass` before `setVSM` will not trigger `work`. Is that intended to have this separation? Alternatively, we could lazily create the `pass` object.

Reply:
to make it clear, this PR has a side effect that the following signal might be emitted before `discoverWork`:

https://github.com/ampproject/amphtml/blob/73f321939d323b74ea588eacad26a4e13becd7a1/src/service/resources-impl.js#L824-L836

L824 can be a no-op as setVSM are chunked.

Reply:
I don't understand. Do you just want to send a fix?

This PR shouldn't change any timing inside of resources with the exception that it no longer guarantees that the first scan happens in the same micro-task as the instantitation of the resource object.